modules/collect/src/main/java/com/opengamma/collect/timeseries/DenseLocalDateDoubleTimeSeries.java

```diff
@@ -0,0 +1,984 @@
+     *
+     * @param startDate  the start date for the series (the value for this
+     *   entry will be stored at position 0 in the array)
+     * @param date  the date to calculate a position for
+     * @return the position in the array where the date would be located
+     */
+    abstract int calculatePosition(LocalDate startDate, LocalDate date);
+
+    /**
+     * Given a start date and a position in an array, calculate what date
+     * the position holds data for.
+     *
+     * @param startDate  the start date for the series (the value for this
+     *   entry will be stored at position 0 in the array)
+     * @param position  the position in the array to calculate a date for
+     * @return the date the position in the array holds data for
+     */
+    abstract LocalDate calculateDateFromPosition(LocalDate startDate, int position);
+  }
+
+  /**
+   * Date corresponding to first element in the array. All other
+   * values can be calculated using date arithmetic to find
+   * correct point
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LocalDate startDate;
+
+  @PropertyDefinition(get = "private", validate = "notNull")
+  private final double[] points;
+
+  /**
+   * Whether we should store data for the weekends (NaN will be stored
+   * if no data is available).
+   */
+  @PropertyDefinition(get = "private", validate = "notNull")
+  private final DenseTimeSeriesCalculation dateCalculation;
+
+  /**
+   * Obtains a time-series from a collection of points.
+   * The points must not contain weekend dates. If weekend
+   * dates are required then use
+   * {@link #of(Collection, DenseTimeSeriesCalculation)}.
+   *
+   * @param points  the list of points
+   * @return the time-series
+   */
+  public static DenseLocalDateDoubleTimeSeries of(Collection<LocalDateDoublePoint> points) {
+    return of(points, SKIP_WEEKENDS);
+  }
+
+  /**
+   * Obtains a time-series from a collection of points
+   * using the specified {@code dateCalculation}.
+   *
+   * @param points  the list of points
+   * @return the time-series
+   */
+  public static DenseLocalDateDoubleTimeSeries of(
+      Collection<LocalDateDoublePoint> points,
+      DenseTimeSeriesCalculation dateCalculation) {
+
+    ArgChecker.notNull(points, "points");
+
+    ArgChecker.isFalse(
+        points.stream().anyMatch(pt ->
+            pt == null || (dateCalculation == SKIP_WEEKENDS && isWeekend(pt.getDate()))),
+        "dates must not contain nulls or weekend dates");
+
+    if (points.isEmpty()) {
+      return EMPTY_SERIES;
+    }
+
+    Pair<LocalDate, LocalDate> extremes = points.stream()
+        .reduce(Pair.of(LocalDate.MAX, LocalDate.MIN), (pair, pt) -> {
+          LocalDate earliest = pair.getFirst();
+          LocalDate latest = pair.getSecond();
+          LocalDate ld = pt.getDate();
+          return Pair.of(ld.isBefore(earliest) ? ld : earliest, ld.isAfter(latest) ? ld : latest);
+        }, (p1, p2) -> p1);
```

Comment:
Simpler and clearer to loop twice. Probably faster as no object creation.

`points.stream().max(dateComparator)`.
`points.stream().min(dateComparator)`.